.github/workflows/wpcloud.yml

```
@@ -0,0 +1,142 @@
+name: WPCloud Unit Testing for WPCOMSH
+
+on:
+  pull_request:
+  push:
+    branches: ['trunk', '*/branch-*']
+concurrency:
+  group: wpcloud-${{ github.event_name }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    name: Install the Monorepo and build wpcomsh
+    runs-on: ubuntu-latest
+    outputs:
+      wpcomsh: ${{ steps.changed.outputs.wpcomsh }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          # codecov.io requires a fetch depth > 1.
+          fetch-depth: 2
+
+      # For pull requests, list-changed-projects.sh needs the merge base.
+      # But it doesn't have to be checked out.
+      - name: Deepen to merge base
+        if: github.event_name == 'pull_request'
+        uses: ./.github/actions/deepen-to-merge-base
+        with:
+          checkout: false
+
+      - name: Setup tools
+        uses: ./.github/actions/tool-setup
+        with:
+          php: ${{ matrix.php }}
+          node: ${{ matrix.node }}
+      - name: Monorepo install
+        run: |
+          echo "::group::Pnpm"
+          pnpm install
+          echo "::endgroup::"
+      - name: Detect if wpcomsh has changed
+        id: changed
+        run: |
+          CHANGED="$(EXTRA=test .github/files/list-changed-projects.sh)"
+
+          # WPCOMSH_CHANGED="$(jq --argjson changed "$CHANGED" -n '$changed | has( "plugins/wpcomsh" ) ')"
+          WPCOMSH_CHANGED="true"
+          echo "wpcomsh=${WPCOMSH_CHANGED}" >> "$GITHUB_OUTPUT"
+      - name: Build wpcomsh
+        if: steps.changed.outputs.wpcomsh == 'true'
+        run: |
+          find . -path ./.github -prune -o -type f -print | sort > /tmp/before.txt
+          echo "::group::Installing and building wpcomsh"
+          pnpm jetpack build -v --deps plugins/wpcomsh
+          echo "::endgroup::"
+
+          # We only want to save the files that were actually created or changed.
+          # But we can't just list them for actions/cache/save, "Argument list too long".
+          # So instead we delete all the unchanged files so we can tell actions/cache/save
+          # to save everything that's left.
+          git -c core.quotepath=off diff --name-only | sort > /tmp/changed.txt
+          if [[ -s /tmp/changed.txt ]]; then
+            grep -F -x -v -f /tmp/changed.txt /tmp/before.txt > /tmp/remove.txt
+          else
+            cp /tmp/before.txt /tmp/remove.txt
+          fi
+          xargs -d '\n' rm < /tmp/remove.txt
+          find . -type d -empty -delete
+
+      - name: Save wpcomsh build cache
+        if: steps.changed.outputs.wpcomsh == 'true'
+        id: wpcomsh-build-cache-save
+        uses: actions/cache/save@v4
+        with:
+          path: |
+            .
+            !./.github/
+          key: ${{ github.sha }}
+  deploy:
+    name: Run PHPUnit on the WPCloud test site
+    runs-on: ubuntu-latest
+    needs: build
+    if: needs.build.changed.outputs.wpcomsh == 'true'
```

Oops, this is why it skipped this time.
```suggestion
    if: needs.build.outputs.wpcomsh == 'true'
```